cmd/cmd.go

```diff
@@ -1,153 +1,199 @@
 package cmd
 
 import (
+	"context"
+	"crypto/tls"
 	"fmt"
-	"os"
-	"path/filepath"
 	"strings"
+	"time"
 
-	"github.com/logrusorgru/aurora"
-	"github.com/mitchellh/go-homedir"
-	ma "github.com/multiformats/go-multiaddr"
-	"github.com/olekukonko/tablewriter"
-	"github.com/spf13/cobra"
+	"github.com/manifoldco/promptui"
 	"github.com/spf13/viper"
-	"google.golang.org/grpc/status"
+	tc "github.com/textileio/go-threads/api/client"
+	"github.com/textileio/go-threads/core/thread"
+	"github.com/textileio/powergate/exe/cli/cmd"
+	bc "github.com/textileio/textile/api/buckets/client"
+	"github.com/textileio/textile/api/common"
+	hc "github.com/textileio/textile/api/hub/client"
+	uc "github.com/textileio/textile/api/users/client"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+var (
+	Timeout = time.Minute
+	Bold    = promptui.Styler(promptui.FGBold)
 )
 
 type Flag struct {
 	Key      string
 	DefValue interface{}
 }
 
-const maxSearchHeight = 50
+type Config struct {
+	Viper  *viper.Viper
+	File   string
+	Dir    string
+	Name   string
+	Flags  map[string]Flag
+	EnvPre string
+	Global bool
+}
 
-func InitConfig(v *viper.Viper, file, cdir, name string, global bool) func() {
-	return func() {
-		found := false
-		pre := "."
-		h := 1
-		for h <= maxSearchHeight && !found {
-			found = initConfig(v, file, pre, cdir, name, global)
-			pre = filepath.Join("../", pre)
-			h++
-		}
-	}
+type ClientsCtx interface {
```

Comment:
Since contexts are needed _within_ `cmd/buck/cli` that can be called from either `hub` or `buck`, each main function implements this interface, which is called by the `Clients` wrapper below.